src/python/WMCore/MicroService/MSOutput/RelValPolicy.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Module to store the RelVal output data placement policy and
+make decisions based on that
+"""
+from __future__ import print_function, division
+
+from copy import deepcopy
+import json
+from WMCore.MicroService.Tools.Common import getMSLogger
+from WMCore.WMException import WMException
+
+
+class RelValPolicyException(WMException):
+    """
+    General exception to be raised when a flaw is found in the RelVal
+    output data placement policy
+    """
+    pass
+
+
+class RelValPolicy():
+    """
+    This module will contain the RelVal output data placement policy, where
+    destinations will be decided according to the dataset datatier.
+    """
+    def __init__(self, policyDesc, listDatatiers, listRSEs, logger=None):
+        self.origPolicy = deepcopy(policyDesc)
+
+        self.logger = getMSLogger(verbose=False, logger=logger)
+
+        self._validatePolicy(policyDesc, listDatatiers, listRSEs)
+        self.dictPolicy = self._convertPolicy(policyDesc)
+
+    def __str__(self):
+        """
+        Stringify this object, printing the original policy
+        """
+        objectOut = dict(originalPolicy=self.origPolicy, mappedPolicy=self.dictPolicy)
+        return json.dumps(objectOut)
+
+    def _validatePolicy(self, policyDesc, validDBSTiers, validDiskRSEs):
+        """
+        This method validates the overall policy data structure, including:
+         * internal and external data types
+         * whether the datatiers exist in DBS
+         * whether the RSEs exist in Rucio
+        :param policyDesc: list of dictionaries with the policy definition
+        :param validDBSTiers: list with existent DBS datatiers
+        :param validDiskRSEs: list with existent Rucio Disk RSEs
+        :return: raise an exception if any validation fails
+        """
+        if not isinstance(policyDesc, list):
+            msg = "The RelVal output data placement policy is not in the expected data type. "
+            msg += "Type expected: list, while the current data type is: {}. ".format(type(policyDesc))
+            msg += "This critical ERROR must be fixed."
+            raise RelValPolicyException(msg) from None
```

Comment:
Once this is raised during the parameter validation process here, is there a place in the `MSOutput` module code to handle this exception, or it is left to be treated as a general exception and just the proper messages to be logged?

Reply:
In case it fails, it's supposed to fail the MSOutput object initialization. Meaning, the service will crash until someone spots the problem.

Reply:
That's fair. 
Thanks